tgui/packages/tgui/interfaces/common/WirePanel/index.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ * @copyright 2023
+ * @author glowbold (https://github.com/pgmzeta)
+ * @license MIT
+ */
+
+import { useBackend } from '../../../backend';
+import { Box, Button, Divider, Section } from '../../../components';
+import { BooleanLike } from 'common/react';
+import type { WirePanelData } from './type';
+import { WirePanelControl, WirePanelCoverStatus } from './const';
+import { Window } from '../../../layouts';
+import { WireList } from './Wire';
+import { IndicatorList } from './Indicator';
+
+export const WirePanel = (props, context) => {
+  return (
+    <Window width={350}>
+      <Window.Content>
+        <Section title="Maintenance Panel"><WirePanelInsert /></Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const WirePanelInsert = (_, context) => {
+  const { act, data } = useBackend<WirePanelData>(context);
+  const { wire_panel } = data;
+  const shouldDisable = getShouldDisable(
+    wire_panel.is_silicon_user,
+    wire_panel.is_accessing_remotely,
+    wire_panel.cover_status,
+    wire_panel.active_controls & WirePanelControl.Silicon
+  );
+  const act_wire = (wire_index: number, action) => {
+    act("actwire", { wire_index, action });
+  };
+
+  return (
+    <Box
+      className="wirePanel_insert"
+      title={`Maintenance Panel${shouldDisable ? ' :: Remote Silicon Access Disabled' : ''}`}
+      open={wire_panel.cover_status === WirePanelCoverStatus.Open}
+      disabled={!!shouldDisable}>
+      <>
+        <WireList wires={wire_panel.wires} act_wire={act_wire} />
+        <Divider />
+        <IndicatorList controls_to_show={wire_panel.controls_to_show} active_controls={wire_panel.active_controls} />
+      </>
+    </Box>
+  );
+};
+
+interface RemoteAccessButtonProps {
+  disabled?: BooleanLike;
+  icon: string;
+  content: string;
+  title: string;
+  onClick: any;
```

Comment:
```suggestion
  onClick: () => void.
```
I realize that the `Button` component isn't TS-ified, but we can be a *little* tidier with what we expect here. As a bonus, if it ever *is* tidied up we'll know if it's compatible with our usages, whereas `any` hides a lot of possible underlying errors.